src/backend/distributed/commands/multi_copy.c

```diff
@@ -3441,7 +3448,15 @@ CopyGetPlacementConnection(ShardPlacement *placement, bool stopOnFailure)
 
 	if (!TransactionConnectedToLocalGroup && placement->groupId == GetLocalGroupId())
 	{
-		TransactionConnectedToLocalGroup = true;
+		/*
+		 * if we are copying into an intermediate file we won't use local copy.
+		 * Files are visible to all transactions so we can still use local execution, therefore
+		 * we shouldn't restrict only using connection in this case.
+		 */
+		if (!isCopyToIntermediateFile)
+		{
+			TransactionConnectedToLocalGroup = true;
```

Comment:
is there a way to move this a bit higher up the call stack to avoid adding the argument?

Reply:
I guess we can move it a bit higher, in that case if there is an error when opening a connection, we will still be setting it. But I guess in that case we will probably abort the transaction so it should be fine.